File: NessieProj/app/src/main/java/org/hacking/nessieproj/Converter.java
package org.hacking.nessieproj;

import android.databinding.InverseMethod;

public class Converter {

    @InverseMethod("toInt")
    public static String toString(Integer value) {
        if (value == null) {
            return "";
        }
        return "" + value;
    }
    public static Integer toInt(String value) {
        return Integer.parseInt(value);
    }

}

File: NessieProj/app/src/main/java/org/hacking/nessieproj/Converter.kt
package org.hacking.nessieproj

import android.databinding.InverseMethod


object Converter{

    fun convertStringToInt(value: String): Int {
        return try {
            Integer.parseInt(value)
        } catch (e: NumberFormatException) {
            -1
        }

    }

    @InverseMethod("convertStringToInt")
    fun convertIntToString(value: Int): String {
        return value.toString()
    }
}